src/Processors/QueryPlan/Optimizations/optimizeReadInOrder.cpp

```diff
@@ -820,6 +820,11 @@ InputOrderInfoPtr buildInputOrderInfo(SortingStep & sorting, QueryPlan::Node & n
             bool can_read = reading->requestReadingInOrder(order_info->used_prefix_of_sorting_key_size, order_info->direction, order_info->limit);
             if (!can_read)
                 return nullptr;
+
+            bool use_buffering = (order_info->limit == 0) && sorting.getSettings().read_in_order_use_buffering;
+            /// Avoid conflict with buffering.
+            if (!use_buffering)
+                reading->enableVirtualRow();
```

Comment:
Buffering is very useful.
Let'sproperly handle virtual row chunks in BufferChunksTransform